client/packages/invoices/src/Returns/DetailView/columns.ts

```diff
@@ -0,0 +1,218 @@
+import {
+  useColumns,
+  //   getRowExpandColumn,
+  //   getNotePopoverColumn,
+  ColumnAlign,
+  GenericColumnKey,
+  SortBy,
+  Column,
+  useCurrency,
+  InvoiceLineNodeType,
+  TooltipTextCell,
+  useColumnUtils,
+  NumberCell,
+} from '@openmsupply-client/common';
+
+interface UseOutboundColumnOptions {
+  sortBy: SortBy<OutboundReturnLine>;
+  onChangeSortBy: (column: Column<OutboundReturnLine>) => void;
+}
+
+// TO-DO: Replace with generated once query is live
+export interface OutboundReturnLine {
+  id: string;
+  itemCode: string;
+  itemName: string;
+  batch: string | null;
+  expiryDate: string | null;
+  numberOfPacks: number;
+  packSize: number;
+  sellPricePerPack: number;
+  type: InvoiceLineNodeType;
+}
+
+// const expansionColumn = getRowExpandColumn<
+//   StockOutLineFragment | StockOutItem
+// >();
+// const notePopoverColumn = getNotePopoverColumn<
+//   StockOutLineFragment | StockOutItem
+// >();
+
+const isDefaultPlaceholderRow = (row: OutboundReturnLine) =>
+  row.type === InvoiceLineNodeType.UnallocatedStock && !row.numberOfPacks;
+
+const getPackSize = (row: OutboundReturnLine) =>
+  isDefaultPlaceholderRow(row) ? '' : row.packSize;
+
+const getNumberOfPacks = (row: OutboundReturnLine) =>
+  isDefaultPlaceholderRow(row) ? '' : row.numberOfPacks;
+
+const getUnitQuantity = (row: OutboundReturnLine) =>
+  isDefaultPlaceholderRow(row) ? '' : row.packSize * row.numberOfPacks;
+
+export const useOutboundReturnColumns = ({
+  sortBy,
+  onChangeSortBy,
+}: UseOutboundColumnOptions): Column<OutboundReturnLine>[] => {
+  const { c } = useCurrency();
+  const { getColumnProperty, getColumnPropertyAsString } = useColumnUtils();
+
+  return useColumns(
+    [
+      //   [
+      //     notePopoverColumn,
+      //     {
+      //       accessor: ({ rowData }) => {
+      //         if ('lines' in rowData) {
+      //           const { lines } = rowData;
+      //           const noteSections = lines
+      //             .map(({ batch, note }) => ({
+      //               header: batch ?? '',
+      //               body: note ?? '',
+      //             }))
+      //             .filter(({ body }) => !!body);
+      //           return noteSections.length ? noteSections : null;
+      //         } else {
+      //           return rowData.batch && rowData.note
+      //             ? { header: rowData.batch, body: rowData.note }
+      //             : null;
+      //         }
+      //       },
+      //     },
+      //   ],
```

Comment:
Have commented out the columns not currently available on the mock query response.

Reply:
Nice one. 

The insert mutation is now merged into the feature branch, so once you pull that in, you could remove the mock API response and query this extra stuff. Maybe in the next PR?